packages/runtime-dom/src/apiCustomElement.ts

```diff
@@ -367,7 +367,7 @@ export class VueElement extends BaseClass {
         const dispatch = (event: string, args: any[]) => {
           this.dispatchEvent(
             new CustomEvent(event, {
-              detail: args
+              detail: args.length === 1 ? args[0] : args
```

Comment:
Could be a breaking change

Reply:
> Could be a breaking change

Yes, whether it is necessary to make this improvement or there is a better way still needs further discussion